active/001-tabular-msa.md

```diff
@@ -0,0 +1,292 @@
+---
+Feature name: tabular-msa
+Start date: 2015-08-14
+RFC PR: 1
+scikit-bio issue: 1053
+---
+
+# Summary
+
+Replace the current experimental `skbio.alignment.Alignment` class with a
+`TabularMSA` class, providing an API for manipulating tabular (row-column)
+multiple sequence alignments.
+
+# Motivation
+
+The current multiple sequence alignment (MSA) class assumes that it is the only
+way to represent an alignment. There are multiple ways of representing MSAs,
+including [partial-order MSAs](http://bioinformatics.oxfordjournals.org/content/18/3/452.short)
+and [A-Bruijn MSAs](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC525693/). The
+reason these cannot be the same object is that a row-column MSA exists as a
+total order (which means that positions exist), while partial-order MSAs do not
+inherently have positions or gaps (because the MSA is represented as a directed
+acyclic graph, or DAG). Thus, while some operations are shared among these MSA
+structures, there are intrinsic differences that necessitate different APIs.
+
+In scikit-bio 0.4.0 the sequence classes were completely redesigned and the
+current MSA class (`Alignment`) makes invalid assumptions about the API of the
+sequences it stores (for example, see [#1036](https://github.com/biocore/scikit-bio/issues/1036)). This gives us an
+opportunity to rework the MSA API to better fit the new sequence classes.
+
+After these changes, we will have an obvious way to introduce new MSA
+structures (e.g., partial-order MSA), better interaction with the new sequence
+API, pandas-style vectorization, and flexible, arbitrary metadata that can
+represent file formats such as
+[Stockholm](http://sonnhammer.sbc.su.se/Stockholm.html).
+
+# Detailed design
+
+A `TabularMSA` object is a mutable, monomorphic, ordered collection of
+`Sequence` objects that are of equal length. As its name implies, `TabularMSA`
+represents a table-like structure where the first dimension is sequences and
+the second dimension is positions in the alignment. Thus, each row corresponds
+to a sequence and each column corresponds to a position. The first dimension is
+optionally indexed by an arbitrary key. Both dimensions are indexable via
+ordinals.
+
+One of the advantages of `TabularMSA` over the existing `Alignment` class is
+vectorization support and boolean indexing. Currently, to filter out alignment
+positions that have a Shannon entropy greater than 0.5:
+
+```python
+aln.subalignment(positions_to_keep=[
+    i for i, v in enumerate(aln.position_entropies()) if v <= 0.5])
+```
+
+With `TabularMSA`:
+
+```python
+aln[:, aln.entropies() <= 0.5]
+```
+
+Another example is filtering out completely-gapped positions. With the current
+API, this will (probably) work:
+
+```python
+aln.omit_gap_positions(1.0 - np.finfo(float).eps)
+```
+
+With `TabularMSA`:
+
+```python
+aln[:, aln.gap_frequencies(relative=True) < 1.0]
```

Comment:
An `axis` should probably be specified, though this (implied) default is nice. See Unresolved Questions [line 276](https://github.com/biocore/scikit-bio-rfcs/pull/1/files#diff-47d603d1ff6048384e4d96c3300116e9R276) for more discussion.